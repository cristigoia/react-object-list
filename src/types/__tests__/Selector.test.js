import React from 'react'
import { shallow } from 'enzyme'
import {
  snapshotTest,
  configureEnzymeAdapter,
} from 'utils/tests'
import { Selector } from '../'


describe('Selector type', () => {
  it('snapshot test', () => {
    snapshotTest(<Selector />)
    snapshotTest(<Selector selected />)
  })
  it('makes a call to the callback function when clicked on', () => {
    const callback = jest.fn()
    const id = 534
    const wrapper = shallow(<Selector toggleSelect={callback} id={id} />)
    expect(callback).not.toBeCalled()
    wrapper.find('input').simulate('click')
    expect(callback).lastCalledWith(id)
  })
})
