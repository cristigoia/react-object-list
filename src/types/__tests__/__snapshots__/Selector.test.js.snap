// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector type snapshot test 1`] = `
<input
  checked={false}
  className="objectlist-table__checkbox"
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`Selector type snapshot test 2`] = `
<input
  checked={true}
  className="objectlist-table__checkbox"
  onChange={[Function]}
  type="checkbox"
/>
`;
